lib/JSON/Validator.pm

```diff
@@ -719,8 +719,12 @@ sub _guessed_right {
 }
 
 sub _is_date_time {
-  $_[0] =~ qr/^(\d+)-(\d+)-(\d+)T(\d+):(\d+):(\d+(?:\.\d+)?)(?:Z|([+-])(\d+):(\d+))?$/io;
+  $_[0] =~ qr/
+    ^(\d+)-(\d+)-(\d+)\D+(\d+):(\d+):(\d+(?:\.\d+)?)   # Date and time
```

Comment:
Omitted? "." means that a character is required, but it can be anything. `\D+` does also not mean it can be omitted. It will match "xxxxx", "x", "T" or a space - not conservative at all to me. I will not take this PR with `\D+`. I think we need to choose between:
* `.`: Exactly one character, which can be anything. (This is how I read the spec)
* `[Tt\s]`: Something that is more sane to me.